features/0067-didcomms-diddoc-conventions/README.md

```diff
@@ -0,0 +1,180 @@
+    - The sender possesses the Did document for the intended recipient(s) of a Did communication message.
+    - The sender has created a [content level message](../../concepts/0021-didcomm-message-anatomy/README.md) that is now ready to be prepared for sending to the intended recipient(s). 
+
+1. The sender resolves the relevant `did-communication` service of the intended recipient(s) Did document.
+2. The sender resolves the recipient keys present in the `recipientKeys` array of the service declaration.
+3. Using the resolved keys, the sender takes the content level message and packs it inside an [encrypted envelope](../../concepts/0021-didcomm-message-anatomy/README.md) for the recipient keys. (*note-2*)
+4. The sender then inspects the `routingKeys` array, if it is found to be empty, then the process skips to step 5. Otherwise, the sender prepares a content level message of type `forward`. The resolved keys from the `recipientKeys` array is set as the contents of the `to` field in the forward message and the encrypted envelope from the previous step is set as the contents of the `msg` field in the forward message. Following this, for each element in the `routingKeys` array the following sub-process is repeated:
+    1. The sender resolves the current key in the routing array and takes the outputted encrypted envelope from the previous step and packs it inside a new encrypted envelope for the current key.
+    2. The sender prepares a content level message of type `forward`. The current key in the routing array is set as the contents of the `to` field in the forward message and the encrypted envelope from the previous step is set as the contents of the `msg` field in the forward message.
+5. Resolve the service endpoint:
+    - If the endpoint is a valid Did URL, check that it resolves to another Did service definition. If the resolution is successful the process from step 2. is repeated using the message outputted from this process as the input message. 
+    - If the service endpoint is not a Did URL, send the message using the transport protocol declared by the URL's scheme.  
+
+> Notes
+>1. There are two main situations that an agent will be in prior to preparing a new message.
+>>-  The agent is responding to a message that has just been received and has the context of the sender key of the previous message.
+>>-  The agent is creating a new message to a connection and will use the default `did-communication` service convention for preparation of a message.
+>With the first case. A targeted lookup of the `did-communication` service definition could be done to find a service definition that features the sender key as a recipient key which would ensure that the response was delivered back to the sender.
```

Comment:
@tplooker can you please insert a blank line above this paragraph? Then it will render separate from the above bullet point.

Also, I don't understand this paragraph clearly - I don't see how the service definition can be identified with a key reference.